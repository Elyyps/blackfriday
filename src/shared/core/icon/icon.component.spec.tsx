import React from "react";
import { shallow } from "enzyme";
import { IconComponent } from "./";
import IconDefault from "@assets/icons/link.svg";

describe("[IconComponent]", () => {
  it("Should render original icon without crashing", () => {
    const renderedComponent = shallow(<IconComponent icon={IconDefault} size="14px" />);
    expect(renderedComponent).toMatchSnapshot();
  });
  it("Should render icon with stroke and strokeWidth without crashing", () => {
    const renderedComponent = shallow(
      <IconComponent icon={IconDefault} strokeColor="red" strokeWidth="2px" size="24px" />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
  it("Should render icon with fill without crashing", () => {
    const renderedComponent = shallow(<IconComponent icon={IconDefault} fillColor="red" size="24px" />);
    expect(renderedComponent).toMatchSnapshot();
  });
  it("Should render icon with fill and stroke without crashing", () => {
    const renderedComponent = shallow(
      <IconComponent icon={IconDefault} strokeColor="red" fillColor="red" size="24px" />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
