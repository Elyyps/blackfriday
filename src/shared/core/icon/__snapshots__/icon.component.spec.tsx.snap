// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[IconComponent] Should render icon with fill and stroke without crashing 1`] = `ShallowWrapper {}`;

exports[`[IconComponent] Should render icon with fill without crashing 1`] = `ShallowWrapper {}`;

exports[`[IconComponent] Should render icon with stroke and strokeWidth without crashing 1`] = `ShallowWrapper {}`;

exports[`[IconComponent] Should render original icon without crashing 1`] = `ShallowWrapper {}`;
